File: hw2/src/shield_test.cpp
#include "shield.h"
#include "gtest/gtest.h"

using namespace std;
using namespace shield;

TEST(Shield, Constructor) {
  string style = "round";
  string material = "steel";
  string color = "silver";
  int weight = 10;
  bool thrown = false;

  Shield shield(style, material, color, weight, thrown);
  ASSERT_EQ(shield.style(), style);
  ASSERT_EQ(shield.material(), material);
  ASSERT_EQ(shield.color(), color);
  ASSERT_EQ(shield.weight(), weight);
  ASSERT_EQ(shield.thrown(), thrown);
}

TEST(Shield, shieldThrow) {
  string style = "round";
  string material = "vibranium";
  string color = "red, white, and blue";
  int weight = 12;
  bool thrown = false;

  Shield shield(style, material, color, weight, thrown);
  ASSERT_EQ(shield.style(), style);
  ASSERT_EQ(shield.material(), material);
  ASSERT_EQ(shield.color(), color);
  ASSERT_EQ(shield.weight(), weight);
  ASSERT_EQ(shield.thrown(), thrown);

  if (shield.thrown == false) {
    print("Captain America throws his mighty shield!");
    shield.throw();
    ASSERT_EQ(shield.thrown(), true);
  }
}

int main(int argc, char** argv) {
  ::testing::InitGoogleTest(&argc, argv);
  return RUN_ALL_TESTS();
}

File: pyArchy/shieldtest.py
import unittest

from shield import Shield

class ShieldTest(unittest.TestCase):
    def testDefaultShield(self):
        style : str = "round"
        color : str = "silver"
        weight : int = 10
        thrown : bool = False

        shield : Shield = Shield(style, color, weight, thrown)
        self.assertEqual(shield.style,style)
        self.assertEqual(shield.color,color)
        self.assertEqual(shield.weight,weight)
        self.assertEqual(shield.thrown,thrown)

    def testThrow(self):
        style : str = "round"
        color : str = "red, white, and blue"
        weight : int = 12
        thrown : bool = False

        shield : Shield = Shield(style, color, weight, thrown)
        self.assertEqual(shield.style,style)
        self.assertEqual(shield.color,color)
        self.assertEqual(shield.weight,weight)
        self.assertEqual(shield.thrown,thrown)

        if shield.thrown == False:
            print("Captain America throws his mighty shield!")
        shield.throw()
        self.assertEqual(shield.thrown, True)
        
if __name__ == '__main__':
    unittest.main()